subprojects/core-api/src/main/java/org/gradle/api/Task.java

```diff
@@ -421,7 +421,10 @@ public String determineName(Task c) {
      * <p>Removes all the actions of this task.</p>
      *
      * @return the task object this method is applied to
+     *
+     * @deprecated Don't use this.
```

Comment:
Maybe we could come up with a better deprecation reason.

Reply:
I don't think we need to sugar-coat this: it is not a useful method, we should just remove it. That said, I'm not against anything if you have a better suggestion.